eventlet/green/ssl.py

```diff
@@ -76,16 +72,21 @@ def __new__(cls, sock=None, keyfile=None, certfile=None,
                         session=kw.get('session'),
                     )
                 else:
-                    ret = _original_wrap_socket(
+                    context = _original_sslcontext(protocol=ssl_version)
+                    context.options |= cert_reqs
+                    if certfile or keyfile:
+                        context.load_cert_chain(
+                            certfile=certfile,
+                            keyfile=keyfile,
+                        )
+                    if ca_certs:
+                        context.load_verify_locations(ca_certs)
+                    if ciphers := kw.get('ciphers'):
```

Comment:
sorry, this is 3.8+ construct, not going to happen anytime soon

Reply:
I know. My aim for this draft was to fix this for 3.12. Maintaining compatibility with previous Python versions was out of scope. If we can determine this approach is *correct*, we can add a giant if around the code. If the walrus would cause a problem even inside an if, no problem removing it.